modules/yugabytedb/yugabytedb.go

```diff
@@ -0,0 +1,246 @@
+		}
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *YugabyteDBContainer
+	if container != nil {
+		c = &YugabyteDBContainer{
+			Container:            container,
+			ysqlDatabaseName:     req.Env[ysqlDatabaseNameEnv],
+			ysqlDatabaseUser:     req.Env[ysqlDatabaseUserEnv],
+			ysqlDatabasePassword: req.Env[ysqlDatabasePasswordEnv],
+		}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("generic container: %w", err)
+	}
+
+	return c, nil
+}
+
+// YSQLConnectionString returns a connection string for the yugabyteDB container
+// using the configured database name, user, password, port, host and additional
+// arguments.
+// Additional arguments are appended to the connection string as query parameters
+// in the form of key=value pairs separated by "&".
+//
+// # Example:
+//
+//	connStr, err := y.YSQLConnectionString(ctx, "sslmode=disable")
+//	if err != nil {
+//		return err
+//	}
+//
+// The above code will return a connection string with the sslmode set to disable.
+func (y *YugabyteDBContainer) YSQLConnectionString(ctx context.Context, args ...string) (string, error) {
+	host, err := y.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	mappedPort, err := y.MappedPort(ctx, ysqlPort)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf(
+		"postgres://%s:%s@%s/%s?%s",
+		y.ysqlDatabaseUser,
+		y.ysqlDatabasePassword,
+		net.JoinHostPort(host, mappedPort.Port()),
+		y.ysqlDatabaseName,
+		strings.Join(args, "&"),
+	), nil
+}
+
+// YCQLConfigureClusterConfig merges the provider [gocql.ClusterConfig] with the
+// yugabyteDB container configuration.
+// The method sets the hosts, keyspace and authenticator fields of the cluster
+// configuration to the values of the yugabyteDB container.
+//
+// # Example:
+//
+//	cluster := gocql.NewCluster()
+//	ctr.YCQLConfigureClusterConfig(ctx, cluster)
+//	session, err := cluster.CreateSession()
+//	if err != nil {
+//		return err
+//	}
+//	defer session.Close()
+//
+// [gocql.ClusterConfig]: https://pkg.go.dev/github.com/gocql/gocql?tab=doc#ClusterConfig
+func (y *YugabyteDBContainer) YCQLConfigureClusterConfig(ctx context.Context, cfg *gocql.ClusterConfig) error {
+	host, err := y.Host(ctx)
+	if err != nil {
+		return err
+	}
+
+	mappedPort, err := y.MappedPort(ctx, ycqlPort)
+	if err != nil {
+		return err
+	}
+
+	inspect, err := y.Container.Inspect(ctx)
+	if err != nil {
+		return err
+	}
+
+	var (
+		keyspace string
+		user     string
+		password string
+	)
+
+	for _, env := range inspect.Config.Env {
+		parts := strings.SplitN(env, "=", 2)
+		switch parts[0] {
+		case ycqlKeyspaceEnv:
+			keyspace = parts[1]
+		case ycqlUserNameEnv:
+			user = parts[1]
+		case ycqlPasswordEnv:
+			password = parts[1]
+		}
+	}
+
+	cfg.Hosts = append(cfg.Hosts, net.JoinHostPort(host, mappedPort.Port()))
+	cfg.Keyspace = keyspace
+	cfg.Authenticator = gocql.PasswordAuthenticator{
+		Username: user,
+		Password: password,
+	}
```

Comment:
question: what happens if these were not found above or are blank?

Reply:
We always have default values set. The only way we could get in a situation where these are empty is if the user explicitly does so by using one of the provided options. 
I'm happy to still check for this situation and return a dedicated error though

Reply:
That sounds like a good idea if the resulting cfg is invalid if they are blank, if blank is valid then I think we can leave as is, what do you think @henripqt ?

Reply:
I've just tested it out using the following which results in a valid session therefore we should not error out on empty configuration values 👍 

```go
func ExampleYugabyteDBContainer_YCQLConfigureClusterConfig() {
	ctx := context.Background()

	yugabytedbContainer, err := yugabytedb.Run(
		ctx,
		"yugabytedb/yugabyte:2024.1.3.0-b105",
		yugabytedb.WithKeyspace(""),  // setting empty keyspace
		yugabytedb.WithUser(""),          // setting empty user
		yugabytedb.WithPassword(""),  // setting empty pwd
	)

	if err != nil {
		log.Printf("failed to start container: %s", err)
		return
	}

	defer func() {
		if err := testcontainers.TerminateContainer(yugabytedbContainer); err != nil {
			log.Printf("failed to terminate container: %s", err)
		}
	}()

	cluster := gocql.NewCluster()
	yugabytedbContainer.YCQLConfigureClusterConfig(ctx, cluster)

	session, err := cluster.CreateSession()
	if err != nil {
		log.Printf("failed to create session: %s", err)
		return
	}

	defer session.Close()

	var i int
	if err := session.Query(`
		SELECT COUNT(*) 
		FROM system_schema.keyspaces 
		WHERE keyspace_name = 'system' // since we don't have a custom keyspace we target the system one 
	`).Scan(&i); err != nil {
		log.Printf("failed to scan row: %s", err)
		return
	}

	fmt.Println(i)
	// Output: 1
}
```